library/package.json

```diff
@@ -57,7 +57,11 @@
     "@typescript-eslint/parser": "^5.60.0",
     "@vitest/coverage-v8": "^0.33.0",
     "eslint": "^8.43.0",
+    "eslint-config-canonical": "^41.3.0",
```

Comment:
Why do we need `eslint-config-canonical` and `eslint-plugin-unicorn`? To me it seems like both are a collection of rules rather than adding specific rules. So I wonder if we really need this as a specific library.

Reply:
`eslint-plugin-unicorn` ships their own rules. For `eslint-config-canonical` we could copy their `regexp` ruleset into our eslint config, but I'm not sure if we should do that

Reply:
Is `eslint-plugin-regexp` not sufficient for regex linting? Do you have an example of useful rules in `eslint-plugin-unicorn` that are missing so far and provide a clear advantage? I don`t want to make the setup too complicated.

Reply:
```
'unicorn/better-regex': 'error', // auto-optimize regexps
'unicorn/prefer-regexp-test': 'error', // RegExp#test is fastest
'unicorn/prefer-string-replace-all': 'warn', // String#replaceAll avoids the need for resetting lastIndex when using cached global regex
'unicorn/prefer-string-starts-ends-with': 'error', // RegExp#startsWith and RegExp#endsWith are faster
```

for `better-regex`, we could use https://github.com/BrainMaestro/eslint-plugin-optimize-regex instead, but at least for my part I like having an eslint config that makes it easy for me and contributors to write efficient code (instead of relying on my memory or on human PR reviewers to catch possible perf improvements); I'd call it "fast by default"

Reply:
In order not to lose control over the plugins, I prefer a clear assignment. One plugin for regex and one for security. On the other hand, these should cover all the important rules, so we don't need to add other plugins for that. In PR #205, for example, we will soon add one for JSDoc.

A compromise for now would be to remove `eslint-config-canonical` and `eslint-plugin-unicorn` and create an issue or PR for the missing rules at `eslint-plugin-regexp`.

Reply:
done